src/components/App/index.jsx

```diff
@@ -0,0 +1,87 @@
+import React, {Component} from 'react';
+import {EndpointEntry} from '../endpointEntry';
+import {Logo} from '../logo';
+import {ImageCard} from '../imageCard';
+import {Spinner} from '../spinner';
+import {ErrorCard} from '../errorCard';
+import './index.less';
+import {poll} from '../../lib/polling.js';
+
+
+class App extends Component {
+
+	constructor() {
+		super();
+		const {host} = window.location;
+		const endpoint = `http://${host}/defaultImages.json`;
+		this.state = {endpoint, loading: true, data: null, error: null};
+	}
+
+	componentWillMount() {
+		const {getData} = this;
+		getData();
+	}
+
+	changeEndpoint = (endpoint) => {
+		const loading = true;
+		const error = null;
+		const {subscription} = this.state;
+		if(subscription){
+			subscription.unsubscribe();
+		}
+		this.setState({endpoint, error, loading}, this.getData);
+	}
+
+	setData = (object) => {
+		const loading = false;
+		const first = Object.keys(object)[0];
+		const data = object[first];
+		this.setState({data, loading});
+	}
+
+	getData = () => {
+		const {setData, setError, state} = this;
+		const {endpoint} = state;
+		const subscription = poll(endpoint).subscribe(setData, setError);
+		this.setState({subscription});
+	}
+
+	setError = (error) => {
+		error = error.toString();
+		const loading = false;
+		this.setState({loading, error});
+	}
+
+	getCards = (imageObject) => {
+		const {_id} = imageObject;
+		return (
+			<ImageCard key={_id} imageObject={imageObject}/>
```

Comment:
if the `ImageCard` component is taking in the key which is a param in `imageObject`, would it still work if you just forward the `imageObject` through?